app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java

```diff
@@ -69,13 +74,12 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
         layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
         newsList.setLayoutManager(layoutManager);
 
-        dataset = new ArrayList<>();
-        newsAdapter = new NewsAdapter(dataset);
-        newsList.setAdapter(newsAdapter);
+        List<News> dataset = getMockData();
```

Comment:
We need to discuss this more, the way I see it, we might need this `List<News> dataset` to be `Observable<List<news>> dataset`.

This is because we don't want the main UI thread to be blocked.

Even if we're going to cache it, we can still return the cached result wrapped in RxJava's `Observable`.